web/docs/migrate-from-0-12-to-0-13.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Migration from 0.12.X to 0.13.X
+---
+
+:::note Are you on 0.11.X or earlier?
+
+This guide covers only the migration from 0.12.X to 0.13.X. If you are migrating from 0.11.X or earlier, please read the [migration guide from 0.11.X to 0.12.X](/docs/migrate-from-0-11-to-0-12.md) first.
+
+:::
+
+## What's new in 0.13.0?
+
+### OAuth providers got an overhaul
+
+In Wasp 0.13.0 we switched away from using Passport for our OAuth providers to using the [Arctic](https://arctic.js.org/) lib from the [Lucia](https://lucia-auth.com/) ecosystem. This change was made to simplify the codebase and to make it easier to add new OAuth providers in the future.
+
+### We added Keycloak as an OAuth provider
+
+If you want to use [Keycloak](https://www.keycloak.org/) as your OAuth provider, you can now use it with Wasp.
+
+## How to migrate?
+
+### Migrate your OAuth setup
+
+We had to make some breaking changes to upgrade the OAuth setup to the new Arctic lib.
+
+Follow the steps below to migrate:
+
+1. **Define the `WASP_SERVER_URL` server env variable**
+
+   In 0.13.0 we introduced a new server env variable `WASP_SERVER_URL` that you need to define. This is the URL of your Wasp server and it's used to generate the redirect URL for the OAuth providers.
+
+   ```bash title="Server env variables"
+   WASP_SERVER_URL=https://your-wasp-server-url.com
+   ```
+
+   In development, we set the `WASP_SERVER_URL` to `http://localhost:3001` by default.
+
+   :::info Migrating a deployed app
+
+    If you are migrating a deployed app, you will need to define the `WASP_SERVER_URL` server env variable in your deployment environment.
+
+    Read more about setting env variables in production [here](./project/env-vars#defining-env-vars-in-production).
+   :::
+
+2. **Update the redirect URLs** for the OAuth providers
+
+    The redirect URL for the OAuth providers has changed. You will need to update the redirect URL for the OAuth providers in the provider's dashboard.
+
+    <Tabs>
+    <TabItem value="before" label="Before">
```

Comment:
I like the before and after stuff, but it seems we lost the javascript/typescript split and this guide therefore only supports typescript?

No big deal, but just FYI to consider whether it makes sense in the future.

Reply:
I was deciding between having the complete matrix [ts, js] x [before, after] or not. It seemed cleaner to include only one of those: most people use Typescript and those using Javascript will get the idea of what needs to change.